### DISCUSSION TITLE
I'm struggling with ssh

### DISCUSSION THREAD

**Discussion by PedroMidueno** [community/community#110409] [Other Feature Feedback, Questions, & Ideas]
### Select Topic Area

Question

### Body

Hi to everyone, i'm not a proficient english speaker so i apologize for any gramar error i could have.

I have 2 accounts in github, i have configured ssh in both of them, i can clone repositories from any of the two accounts without problems, but nos i'm working in a personal project and i'm trying to push my repo with ssh but i'm havin an error and i have not been able to sove it, i'lll add some screenshots to show what is it about.

![image](https://github.com/community/community/assets/113569016/317690b1-3091-4b80-aa40-3dd82436f6b3)
Here i cloned a repo with ssh and it was done succesfully.

![image](https://github.com/community/community/assets/113569016/ac10993f-9e5f-413a-a249-bb120465b1be)
I have added remote origin in other repo with ssh too, this repo is not in github yet.

![image](https://github.com/community/community/assets/113569016/8c650700-612f-4fd7-981f-d574069655b1)
But when i try to push i see this error and i have tried to solve it but i haven't been able to do it.

![image](https://github.com/community/community/assets/113569016/2b86417f-36a0-452e-8ed6-464eff7ed4b0)
This is the repo where i'm trying to push

If someone can help me to solve it i wuold appreciate it so much.

**Comment by gil-son** [comment]
Probably its happen because your computer has just a credential of just one account

To check it is necessary going to Credentials of web:

Control Painel -> User Accounts -> Credential Manager -> Search by your Github account

Pay attention! You have to remove that account to be able to configure the user.name and the email.name of the other account

But you can configure the multiplies accounts if you use [SSH Key](https://gist.github.com/rahularity/86da20fe3858e6b311de068201d279e3)


If I supported you consider to mark my answer. God bless you!

**Comment by 0xSecureByte** [comment]
# GitHub Authentication with Multiple Accounts

If you have multiple GitHub accounts, SSH keys might be difficult to setup and you can use personal access tokens for authentication, you can store the tokens as environment variables. This allows you to manage authentication for different accounts more securely.

## 1. Generate Personal Access Tokens

- Go to your GitHub account settings.
- Navigate to "Developer settings" -> "Personal access tokens."
- Generate a token for each account, ensuring you grant the necessary scopes.

## 2. Store Tokens in Environment Variables

1. Open your shell profile configuration file (e.g., `~/.bashrc`, `~/.zshrc`, or `~/.bash_profile`) using a text editor.

2. Add the following lines for each GitHub account:

   ```bash
   export GIT_ACC1=your_first_token_here
   export GIT_ACC2=your_second_token_here

## 3. Update Remote URLs

For each Git repository, update the remote URL using the corresponding environment variable:

```bash
git remote set-url origin https://<username>:$GIT_ACC1@github.com/<username>/<repo-name>.git
```
Replace "username" with your github account username and "repo-name" with your repository name

Example:
```bash
git remote set-url origin https://PedroMidueno:$GIT_ACC1@github.com/PedroMidueno/rick_and_morty.git
```
## 4. Push to repositories

**Consider marking this answer if it helps** 💖

**Reply by sherwind-tsm** [reply]
Is the an ssh version of these solution?

**Reply by 0xSecureByte** [reply]
Yes, by adding two different ssh keys in the corresponding accounts
